client/sheepshead/SharedSheep/Trick/Trick.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using SharedSheep.Player;
+using SharedSheep.Card;
+
+namespace SharedSheep.Trick
+{
+    public class Trick : ITrick
+    {
+
+        public List<(IPlayer, ICard)> TrickCards { get; private set; }
+
+
+
+        public Trick()
+        {
+
+        }
+        public Trick(List<(IPlayer, ICard)> cards)
+        {
+            TrickCards = new List<(IPlayer, ICard)>(5);
+            TrickCards = cards;
+        }
+
+        
+        public void AddCardAndPlayer(IPlayer player, ICard card)
+        {
+            if(TrickCards.Count < 5)
+            TrickCards.Add((player,card));
+        }
+
+        public ICard LeadingCard()
+        {
+            return TrickCards[0].Item2;
+        }
+
+
+        public IPlayer TheWinnerPlayer()
+        {
+            ICard card ;
+            IPlayer winner = TrickCards[0].Item1;
+            for (int i=0; i<TrickCards.Count;i++)
```

Comment:
I would suggest that instead of looping in both winner methods, maybe create a helper method that returns the winner tuple, and each of these just calls that and returns the appropriate item.

Reply:
Or, we could have a private member that keeps track of the winner every time a card is added. Then we don't need to loop at all.